### PR TITLE
Update warning message for containers package detection when building SDK-containerizable project types

### LINKED ISSUE #39931: `Microsoft.NET.Build.Containers` version '8.0.202' warning is incorrect on `8.0.1xx` SDK
I'm using .NET 8 from the Ubuntu archive on Ubuntu 24.04.

```bash
$ dotnet --version
8.0.102
$ dotnet add package Microsoft.NET.Build.Containers
$ dotnet publish -t:PublishContainer -p:EnableSdkContainerSupport=true
MSBuild version 17.8.5+b5265ef37 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
  dotnetapp -> /home/rich/git/dotnet-docker/samples/dotnetapp/bin/Release/net8.0/dotnetapp.dll
/home/rich/.nuget/packages/microsoft.net.build.containers/8.0.202/build/Microsoft.NET.Build.Containers.targets(194,5): warning : Microsoft.NET.Build.Containers NuGet package is explicitly referenced. Consider removing the package reference to Microsoft.NET.Build.Containers as it is now part of .NET SDK. [/home/rich/git/dotnet-docker/samples/dotnetapp/dotnetapp.csproj]
  dotnetapp -> /home/rich/git/dotnet-docker/samples/dotnetapp/bin/Release/net8.0/publish/
  Building image 'dotnetapp' with tags 'latest' on top of base image 'mcr.microsoft.com/dotnet/runtime:8.0'.
  Pushed image 'dotnetapp:latest' to local registry via 'docker'.
```

That's not the best, right? So, either the package should only emit that warning with `8.0.200`+ or we should have guidance to use an `8.0.1xx` package for an `8.0.1xx` SDK (AKA "matching").

### PR DESCRIPTION
**Pull request by baronfel**
## Description

We need to make the warning about removing the Microsoft.NET.Build.Containers NuGet package only fire when running in the context of an SDK that can actually package the kind of app being packaged (since this is still being pushed as a PackageReference as well as bundled into the SDK for certain project types).

Fixes #39931.

## Customer Impact

Customers only get a warning if the SDK they are building containers with is actually capable of natively containerizing the kind of project they are building - this means that 8.0.100 SDK users (e.g. Linux distro installs) will only see this error for Web and Worker projects, never console apps. This is valuable because those users will rarely have access to an SDK that _can_ natively target console apps during the 8 cycle, which is LTS.

## Regression

**Yes**, this warning should not have been triggering for different kinds of projects in the different SDK bands - we have been errantly throwing this in different ways since 7.0.400.

## Risk

**Low**, this makes a warning less likely to fire by removing the warning in situations where it is not applicable

## Testing

Automated tests cover these scenarios already.

We should backport this to 8.0.100 so that the LTS SDKs do not throw errant warnings.

### CODE REVIEW

**Comment by baronfel** [comment]
Per @marcpopMSFT we're locked down this week so holding this one until next week for tactics approval.

File: src/Containers/packaging/build/Microsoft.NET.Build.Containers.targets
Comment (rainersigwald) [inline src/Containers/packaging/build/Microsoft.NET.Build.Containers.targets:214]: Can this simplify to

```suggestion
      <_ContainerPackageIsPresent Condition="'@(PackageReference->AnyHaveMetadataValue('Identity', '$(_ContainersPackageIdentity)')) != ''">true</_ContainerPackageIsPresent>
```

(and then maybe inline it?)

File: src/Containers/packaging/build/Microsoft.NET.Build.Containers.targets
Comment (rainersigwald) [inline src/Containers/packaging/build/Microsoft.NET.Build.Containers.targets:214]:
```suggestion
      <_ContainerPackageIsPresent Condition="'@(PackageReference->AnyHaveMetadataValue('Identity', '$(_ContainersPackageIdentity)'))">true</_ContainerPackageIsPresent>
```

**Review by rainersigwald — APPROVED** [review]

File: src/Containers/packaging/build/Microsoft.NET.Build.Containers.targets
Comment (baronfel) [reply src/Containers/packaging/build/Microsoft.NET.Build.Containers.targets:225]: It should - I think one of the hurdles there was that we couldn't necessarily use the .NET SDK repo's localization message Task for this and so we had punted for MSBuild-delivered logs. I can look into this for 9.x though.

**Comment by baronfel** [comment]
Approved over email.

**Pull request merged**
